### PR TITLE
PDCalibration sanity checks

### PR DESCRIPTION
**Pull request by peterfpeterson**
There was a report from users about how well PDCalibration worked and did not protect the user from meaningless results. This is a big step forward in making it better.

**To test:**

Try calibrating a powder instrument with more parameters turned on. There will not be as many results that supply `DIFA` or `TZERO`. There are also additional parameters for limiting the range of those two parameter results.

*There is no associated issue*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/PDCalibration.cpp [hunk 1]
```diff
@@ -272,6 +305,38 @@ void PDCalibration::exec() {
...
+    m_tzeroMax = tzeroRange[1];
+
+    std::stringstream msg;
+    msg << "Using tzero range of " << m_tzeroMin << " <= "
+        << "TZERO <= " << m_tzeroMax;
+    g_log.information(msg.str());
+  } else {
+    g_log.information("Using all TZERO values");
+
+    m_tzeroMin = -1. * std::numeric_limits<double>::max();
```
Comment (rosswhitfield) [inline Framework/Algorithms/src/PDCalibration.cpp]: `std::numeric_limits<double>::lowest()`?

Comment (peterfpeterson) [inline Framework/Algorithms/src/PDCalibration.cpp]: I didn't notice that one in the documentation. I knew that `std::numeric_limits<double>::min()` gave back a small positive number which wasn't what I wanted.

**Comment by rosswhitfield** [comment]
Works as described including the new range options. 
LGTM :shipit:

File: Framework/Kernel/inc/MantidKernel/Diffraction.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
+#ifndef MANTID_KERNEL_DIFFRACTION_H_
+#define MANTID_KERNEL_DIFFRACTION_H_
+
+#include "MantidKernel/DllConfig.h"
+
+namespace Mantid {
+namespace Kernel {
+namespace Diffraction {
+
+/** Diffraction : TODO: DESCRIPTION
```
Comment (ianbush) [inline Framework/Kernel/inc/MantidKernel/Diffraction.h]: Description?

File: docs/source/release/v3.10.0/diffraction.rst [hunk 3]
```diff
@@ -21,6 +21,7 @@ Powder Diffraction
 - Bugfix in :ref:`SNAPReduce <algm-SNAPReduce>` with loading previous normalizations
 - :ref:`SNSPowderReduction <algm-SNSPowderReduction>` now supports splitters in format of ``MatrixWorkspace`` and general ``TableWorkspace``.
 - A new NOMAD instrument definition file with corrected values.
+- :ref:`algm-PDCalibration <algm-PDCalibration>` is better at giving out physically meaningful results.
```
Comment (ianbush) [inline docs/source/release/v3.10.0/diffraction.rst]: Can this be elaborated?

**Pull request merged**